neo4j/src/main/java/org/hibernate/ogm/datastore/neo4j/dialect/impl/Neo4jAssociationSnapshot.java

```diff
@@ -63,6 +68,28 @@ public int size() {
 	}
 
 	private static Iterable<Relationship> relationships(Node ownerNode, AssociationKey associationKey, String relationshipType) {
+		if ( relationshipType.contains( "." ) ) {
```

Comment:
What is the actual semantics of that. Maybe push the condition to method with a descriptive name (think you even did it at another place).